### ISSUE TITLE
False positive: reportUnnecessaryComparison

### ISSUE THREAD

**Issue by Questionsdenombres** [microsoft/pyright#5339]
## Environment data
- Windows 11
- VScode 1.79.2
- Python 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)] on win32
- Pylance v2023.6.20 (mode strict)
## Code Snippet
```python
N: int = 10
tags: list[bool] = [False for _ in range(N)]
adjacents: list[list[int]] = [[] for _ in range(N)]

def walk(start_node: int):
    step: int = 1
    node: int = start_node
    while True:
        for node in adjacents[node]:
            if (not tags[node]) or ((node == start_node) and (step == 2)): # here is the report
                tags[node] = True
                step += 1
                break
        else:
            return
```
## Actual behavior
- A report is displayed under the code: `step == 2`
- reportUnnecessaryComparison: Condition will always evaluate to False since the types "Literal[1]" and "Literal[2]" have no overlap
## Expected behavior
- no report. 
- It seems that pylance understands that the `step' variable will always evaluates Literal[1].
- The code meaning is irrelevant: just look at the `step` variable. The condition is part of an `or` and the variable is incremented two lines later within a loop.

**Comment by erictraut** [comment]
This will be fixed in the next insiders release of pylance.

**Comment by erictraut** [comment]
This is included in pyright 1.1.315, which I just published. It will also be included in this week's insiders release of pylance.
